hbase-shell/src/main/ruby/hbase/admin.rb

```diff
@@ -114,6 +114,9 @@ def compact(table_or_region_name, family = nil, type = 'NORMAL')
     #----------------------------------------------------------------------------------------------
     # Switch compaction on/off at runtime on a region server
     def compaction_switch(on_or_off, regionserver_names)
+      unless /true|false/i.match?(on_or_off)
+        raise ArgumentError, 'compactionSwitch first argument only accepts "true" or "false"'
```

Comment:
`compactionSwitch first argument  ...`   -> `compaction_switch first argument ...`,    would this be better?

Reply:
Yes this would be better, thx